### ISSUE TITLE
Interact with the followers of given list of users

### ISSUE THREAD

**Issue by timgrossmann** [InstaPy/InstaPy#444]
> This is the first Issue that will get a bounty on it. (More to come)
This has a high priority.
[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=47675897)](https://www.bountysource.com/issues/47675897-interact-with-the-followers-of-given-list-of-users?utm_source=47675897&utm_medium=shield&utm_campaign=ISSUE_BADGE)

The idea is to interact with the followers of a given account. This should be implemented so that I can provide a list of accounts and a number of followers I want to interact with.

E.g. I provide the feature with the list of `['natgeo', 'geo']`, I then want the script to go to these accounts and interact with e.g. 50 of their followers. (Following is highest prio)
The followers can either chosen randomly or simply the top X followers.

>In addition, it would be great to allow the liking+commenting features of InstaPy to interact with this specific “followers of specific users” list as well.

This could looks something like this:

```python
.follow_from_account(['natgeo', 'geo'], amount=50, random=True) 
```

**DoD**:
- I can provide a list of users from which the followers will be followed
- I can provide a number of followers which will be used for each provided account
- I can either choose to do it top-down or "random", which will simply follow some random followers
- The feature can be "easily" extendable with the next "interact with user" feature

**Comment by gitpatrickhub** [comment]
Not sure this is open to additions/feedback, but just in case.

> In addition, it would be great to allow the liking+commenting features of InstaPy to interact with this specific “followers of specific users” list as well."

Support for the following behavior would be great;
For specified IG account scroll follower list for specified number, like 3 most recent posts from each follower and post a comment on 33% and follow 10% --> result for your example would be for 50 random natgeo followers --> 150 posts liked (3 posts per 50 followers) --> 50 posts commented --> 5 accounts followed

**Comment by timgrossmann** [comment]
@gitpatrickhub Yes it is, I just wanted to make several smaller issues instead of one big one.
I think the "interaction" with other users should be a separate method that can then simply be called in this method.

Good to see people interested!

**Comment by bardascat** [comment]
+1 this would be a great functionality..

**Comment by bardascat** [comment]
Also it would be nice to automatically interact with followers of somebody who you are following...

**Comment by jeffreylm** [comment]
Didn't realize that this is what this function does. What would make this function more valuable is if it follows the people that Like or that commented on  a given post or last post.

Those people are the ones you really want, you want highly active followers. My experience is when you follow them, they follow you back, then they become highly active on your own feed, because that is just there nature.

**Feature recap - From specified accounts**
Follow people of last post who commented
Follow people of last post who like the post

Set these follows as special users  (they have a higher value) that you don't want unfollowed unless given as an arguemnt to add the the unfollow pool.

People followed under those conditions are tagged in the database as high value. That way the unfollow feature need a value of true for High value unfollow to be able to unfollow them.

I really wish I knew how to program, I would have been helping to implement some of these ideas in your project.

Thank you for making such a helpful piece of software.

**Comment by jeffreylm** [comment]
One question about the line ".follow_from_account(['natgeo', 'geo'], amount=50, random=True) ".
Am I to understand that htis feature isn't implemented in this way as of yet?

**Comment by timgrossmann** [comment]
@jeffreylm yes

**Comment by jeffreylm** [comment]
@timgrossmann

Ok, then it is clear. And what do you think of my version of the feature being added to your script? Would you put it under the bounty?

**Comment by sojan-official** [comment]
guys i found InstaPy yesterday and its a neatly written piece of software and I would love to contribute.
If no one else is working on this already, i will pick this up.

**Comment by jeffreylm** [comment]
@sojan-official That would be amazing!!!

Will you also incorporate the ideas I gave?

**Comment by ahmadudin** [comment]
@timgrossmann  Is this boutny still on? i actually have made these requested function and used it for a week. Btw It got me good follower gain.

**Comment by jeffreylm** [comment]
@ahmadudin Does your feature set include my features also?

**Comment by timgrossmann** [comment]
@ahmadudin Yes it's still on

Just make sure that the given requirements are met 😉

**Comment by jeffreylm** [comment]
@ahmadudin What also is important is that you follow the active users of an account, the rest are passive and will most probaly be passive on your account also. So in a sense Tim's original request doesn't defrenciate and will get you there passive and active by chance, but adding the feature to also focus down on the commenters & likers of that account, makes for a higher quality follow list.

And that speaks also to the fact that we are also only allowed to follow so many per day, makes this aspect also interesting.

**Comment by jeffreylm** [comment]
@timgrossmann does my features also fall under your bounty?

**Comment by ahmadudin** [comment]
@timgrossmann okay then
@jeffreylm for now its just following from followers pool, but i also have something in my mind that similiar to your feature. and for following the likers i dont think its possible for now, because we dont have access for that in instagram web.

**Comment by BernardoGO** [comment]
I think #477 meets all the requirements. There is some extra functionality like following the followers and the users that an account follows since I only saw this issue/bounty after I did the pull request.

**Comment by ahmadudin** [comment]
Check #479, works simply as required

**Comment by jeffreylm** [comment]
@BernardoGO Is 477 now merged? I am still getting use to GITHUB, I can't tell what is now included. I am seeing you and @ahmadudin talk about the differences, but has @timgrossmann added any of your version?

How do I read if it is so. I watched a few hours of tutorials thus so far, but still haven't gotten my head around it. I do understand the pull concept, the fork concept. I cloned the Tim version to my drive the first day I came here and made some changes to the quickstart file then the changes to the login file according to Bernado, and I now get the impression that Tim has accepted his changes, but I can't decipher.

Can I get a clear answer on this and if 477, 478, 479 are now merged into the Master of not?

Thank you in advance,
Jeffrey

**Comment by jeffreylm** [comment]
@BernardoGO & @ahmadudin It seems like you to have make the same feature with varying differences.
@ahmadudin  when you say "works simply as requited", which one of the many comments of that are you refering to? Being that I had an extended version and Tim had his bounty version.

Thank you in advance.

**Comment by BernardoGO** [comment]
@jeffreylm #477 is not merged yet.
You can tell if it's merged, closed or open by opening the pull request. If it says it's open it's not merged yet, you can see the green icon.
If it's closed it won't be merged. 
The purple icon means that it was merged, like #476.

If you want to see what has changed you can open the pull request and click on the last tab: changed files. If you want to test it just clone of download the fork repo in my profile. For #477 use the branch "follow".

#479 is not merged.
#478 is closed, not merged.
#477 is not merged.

**Comment by timgrossmann** [comment]
Fixed by #477

**Comment by jeffreylm** [comment]
@BernardoGO Really thank you for that explanation.
No one in the videos would just explain these simple concepts. This is truly the basics, now I can follow whats going on here with the merges.


One more question. In the same way you explained above.
Is there is simple workflow with github desktop to have new changes overwrite my local copy?
I got messages about stashing and commiting before I get the updated version from online. 

Saw that #477 is now merged since your last message. 

Greetings,
jeffrey
